flocker/node/agents/ebs.py

```diff
@@ -559,20 +603,33 @@ def attach_volume(self, blockdevice_id, attach_to):
                     # UserGuide/device_naming.html), wait for new block device
                     # to be available to the OS, and interpret it as ours.
                     # Wait under lock scope to reduce false positives.
-                    new_device = _wait_for_new_device(blockdevices,
-                                                      volume.size)
+                    device_path = _wait_for_new_device(
```

Comment:
I was kinda hoping to just delete the whole code path with the lock, since it could erroneously notice legit EBS devices attached by some other system. But I guess that failure mode would now mostly just result in a retry... so OK that seems fine for now.